### PR TITLE
Feature/UI lib create modal variant based on blueprint new modals UI cu 2ad6kf0

### PR DESCRIPTION
**Pull request by lcscou**
- [x] Have you made sure your branch is up to date with it's parent branch before creating this PR?
- [x] Are all warnings and errors fixed?
- [x] Have you throughoutfully commented your changes?
- [x] Have you added proper jsdocs comments?
- [x] Have you updated the **CHANGELOG** file?
  - [x] Have all _**breaking changes**_ been properly documented and communicated?
  - [x] Have all _changes_ been properly documented and communicated?
  - [x] Have all _fixed issues_ been properly documented and communicated?
- [x] **Have you raised the `version` in package.json if applicable?**
- [x] Have you randomly paid a compliment to a colleague today?

---

- [x] **Do you agree that by pushing to master(if the case). You'll generate a new package build?**

### CODE REVIEW

**Comment by yanikproulx** [comment]
Task linked: [CU-2ad6kf0 [UI lib] Create modal variant based on blueprint new modals UI](https://app.clickup.com/t/2ad6kf0)

File: src/components/Modal/index.tsx [hunk 1]
```diff
@@ -16,13 +16,25 @@ export type ModalProps = {
...
    */
   title?: string;
+  /**
+   * Whether or not close button should be displayed
+   */
+  isClosable?: string;
+  /**
+   * What style of the componenteZ should be displayed
+   */
+  variant?: 'Default' | 'New';
```
Comment (adarleyjrr) [inline src/components/Modal/index.tsx:33]: `variant?: 'default' | 'new';`

File: src/components/Modal/index.tsx [hunk 2]
```diff
@@ -16,13 +16,25 @@ export type ModalProps = {
...
-   * If yes. It won't push the content down and the padding should be set on the children component.
+   * If yes. It won't push the content down and the hasPadding should be set on the children component.
    */
   isHeaderAbsolute?: boolean;
   /**
    * A title string
    */
   title?: string;
+  /**
+   * Whether or not close button should be displayed
```
Comment (adarleyjrr) [inline src/components/Modal/index.tsx:27]: Please update this comment to also specify that modal is not closable if this is set.

File: src/components/Modal/index.tsx [hunk 3]
```diff
@@ -16,13 +16,25 @@ export type ModalProps = {
...
+   */
+  isClosable?: string;
+  /**
+   * What style of the componenteZ should be displayed
+   */
+  variant?: 'Default' | 'New';
+  /**
+   * Whether the content container should be removed
```
Comment (adarleyjrr) [inline src/components/Modal/index.tsx:35]: I don't think this is what the prop does, please redo this comment.

File: src/components/Modal/index.tsx [hunk 4]
```diff
@@ -39,13 +51,19 @@ export type ModalProps = {
  * @param props.title A title string
  * @param props.actions A component that if provided will add a "actions" footer
  * @param props.children The provided children content
+ * @param props.isClosable Whether or not modal can be closed
+ * @param props.variant What style should be displayed
+ * @param props.removePadding Whether content container should be removed
```
Comment (adarleyjrr) [inline src/components/Modal/index.tsx:56]: Update these guys too when you do the updates above.

File: src/components/Modal/index.css [hunk 5]
```diff
@@ -42,6 +43,10 @@
     /* Adds the default border radius */
     /* Animate the fadeUp effect */
     @apply relative shadow-xl my-0 mx-auto md:mt-14 bg-white max-w-full min-w-screen-90 md:min-w-initial pui-border-radius pui-animate-fadeUp;
+    /* Define a border radius that matches with new design */
+    &.New {
```
Comment (adarleyjrr) [inline src/components/Modal/index.css:47]: `&.new`

File: src/components/Modal/index.tsx [hunk 6]
```diff
@@ -82,20 +100,22 @@ export const Modal: React.FC<ModalProps> = ({
...
     <ReactPortal selector="#modal-root">
       <div className={classnames('pui-modal', { open: isOpen })}>
-        <div className="pui-modal-container">
-          <div className={classnames('pui-modal-header', { 'absolute z-30': isHeaderAbsolute })}>
-            <h3>{title}</h3>
-            <button className="pui-btn-icon text-pui-paragraph-900 p-4" onClick={onToggle}>
-              {/* Adds a close icon */}
-              <CrossIcon className="pui-animate-scaleHover-target" />
-            </button>
+        <div className={classnames('pui-modal-container', variant)}>
```
Comment (adarleyjrr) [inline src/components/Modal/index.tsx:108]: `<div className={classnames('pui-modal-container', { 'new': variant === 'new' })}>`

**Review by adarleyjrr — CHANGES_REQUESTED** [review]
Just a few notes, thank you!

File: src/components/Modal/index.tsx [hunk 7]
```diff
@@ -39,13 +51,19 @@ export type ModalProps = {
...
+ * @param props.removePadding Whether the padding of content container should be removed
  */
 export const Modal: React.FC<ModalProps> = ({
   onToggle,
   isOpen: isOpenProps,
   isHeaderAbsolute,
   title,
   actions,
+  removePadding,
+  variant = 'Default',
```
Comment (adarleyjrr) [inline src/components/Modal/index.tsx:65]: Please replace for `variant = 'default'`. Is your TypeScript running? That assignment should have raised an error.

File: src/components/Modal/index.tsx [hunk 8]
```diff
@@ -16,13 +16,25 @@ export type ModalProps = {
   onToggle?: () => void;
   /**
    * Whether or not the top section with the close button should be absolute or not.
-   * If yes. It won't push the content down and the padding should be set on the children component.
+   * If yes. It won't push the content down and the hasPadding should be set on the children component.
```
Comment (adarleyjrr) [inline src/components/Modal/index.tsx:19]: `hasPadding` does not exist anymore. I think this comment should remain `If yes. It won't push the content down and the padding should be set on the children component.`? I am not sure, please just make it clear according to the actual behaviour after your changes were made.

**Review by adarleyjrr — APPROVED** [review]
LGTM!

**Pull request merged**